### ISSUE TITLE
Generated select and checkbox inputs on form_components not working

### ISSUE THREAD

**Issue by leandrocp** [phoenixframework/phoenix#4998]
### Environment

* Elixir version (elixir -v): 1.14 or 1.15-dev
* Phoenix version (mix deps): dev (master)

### Steps to reproduce

```
cd phoenix/installer
mix phx.new phx_blog --dev --install
cd phx_blog
mix phx.gen.live Blog Post posts title body:string public:boolean status:enum:unpublished:published:deleted
# follow instruction to add routes
mix ecto.drop; mix ecto.setup 
iex -S mix phx.server
```

- Navigate to http://localhost:4000/posts/new, choose a select option or check "enabled"
- Input some text on title or body

The previous changed field (select or checkbox) should keep its value but it's reseted to the default value when changing other fields.

**Comment by shamanime** [comment]
I'll try to fix this.

**Comment by shamanime** [comment]
Select is fixed on #5000.

**Comment by josevalim** [comment]
Closing in favor of the PR!
